src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -538,7 +539,7 @@ public override int Read (byte [] buffer, int offset, int count)
 				if (d.Position == d.Length) {
 					lock (dataLock) {
 						// this is the same object, it was done to make the cleanup
-						data.Dequeue ();
+						data.Dequeue ().Dispose ();
 						current?.Dispose ();
```

Comment:
also `Dispose` was already called here: no real harm in calling twice, but it was not missing.
This is now (at best) unbalanced so no native instances are freed.